### PR TITLE
Add safe-to-evict annotation to pod-template-file

### LINKED ISSUE #37351: safe-to-evict annotation is not set when using KubernetesExecutor
### Official Helm Chart version

1.11.0 (latest released)

### Apache Airflow version

2.8.0

### Kubernetes Version

1.29

### Helm Chart configuration

```yaml
executor: KubernetesExecutor
workers:
  safeToEvict: false
```

### Docker Image customizations

_No response_

### What happened

When running Airflow with `KubernetesExecutor`, the `workers.safeToEvict` configuration is not being used, as this field is only read in the workers deployment (used by `CeleryExecutor`).

### What you think should happen instead

_No response_

### How to reproduce

1. Deploy Airflow using Kubernetes Executor. Set `workers.safeToEvict: false` in the chart
2. Run a DAG
3. Check worker pods annotations and see that the safe-to-evict one is missing

### Anything else

Related PR: https://github.com/apache/airflow/pull/35130

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by csp33**
closes: #37351 

# safeToEvict = false
image

# safeToEvict = true
image

### CODE REVIEW

**Comment by csp33** [comment]
Thinking about it, I can't find a reason why this field should be set to `true` when using KubernetesExecutor. Maybe we can hardcode it to be always false in the pod template file.
WDYT?
@jedcunningham

**Comment by dirrao** [comment]
> Thinking about it, I can't find a reason why this field should be set to `true` when using KubernetesExecutor. Maybe we can hardcode it to be always false in the pod template file.
> WDYT?
> @jedcunningham

This should be configurable instead of hardcoding. May be default can be true. When it is set to true, It allows pods to evict for node maintenance or node auto-scaling down.

**Review by dirrao — CHANGES_REQUESTED** [review]
Static doc checks are failing. Can you check and fix them?

**Comment by hussein-awala** [comment]
I'm -0.5 here, this will work only with kubernetes/autoscaler project but not with the other autoscalers (e.g karpenter)

**Comment by csp33** [comment]
@dirrao 
static checks fail becuase of the master branch of the cluster-autoscaler repo link 
image

what can I do?

**Comment by csp33** [comment]
@hussein-awala 
yep, this would only add the safe-to-evict annotation, which refers to k8s official cluster autoscaler. If anyone wants to use Karpenter, the associated annotation should be added manually

**Comment by tsgit** [comment]
this leads to duplicate annotation when rendering the helm chart.
```
$ helm repo update apache-airflow
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "apache-airflow" chart repository
Update Complete. ⎈Happy Helming!⎈
$ helm template airflow apache-airflow/airflow > airflow-12.yaml
$ grep -B12 evict airflow-12.yaml
....
    metadata:
      labels:
        tier: airflow
        component: scheduler
        release: airflow
      annotations:
        checksum/metadata-secret: 0ad002abeb498ee01d66ab45091eb511f65f40b914c7129fcdfc23880421fc6a
        checksum/result-backend-secret: 98a68f230007cfa8f5d3792e1aff843a76b0686409e4a46ab2f092f6865a1b71
        checksum/pgbouncer-config-secret: 1dae2adc757473469686d37449d076b0c82404f61413b58ae68b3c5e99527688
        checksum/airflow-config: dcdc0a128752f6f8115c12e3664947ed5f9c5d5080cb8065192ccebc04c06324
        checksum/extra-configmaps: e862ea47e13e634cf17d476323784fa27dac20015550c230953b526182f5cac8
        checksum/extra-secrets: e9582fdd622296c976cbc10a5ba7d6702c28a24fe80795ea5b84ba443a56c827
        cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
        cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
.......

  template:
    metadata:
      labels:
        tier: airflow
        component: triggerer
        release: airflow
      annotations:
        checksum/metadata-secret: 0ad002abeb498ee01d66ab45091eb511f65f40b914c7129fcdfc23880421fc6a
        checksum/pgbouncer-config-secret: 1dae2adc757473469686d37449d076b0c82404f61413b58ae68b3c5e99527688
        checksum/airflow-config: dcdc0a128752f6f8115c12e3664947ed5f9c5d5080cb8065192ccebc04c06324
        checksum/extra-configmaps: e862ea47e13e634cf17d476323784fa27dac20015550c230953b526182f5cac8
        checksum/extra-secrets: e9582fdd622296c976cbc10a5ba7d6702c28a24fe80795ea5b84ba443a56c827
        cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
        cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
.....
```

**Comment by csp33** [comment]
hi @tsgit , this behavior is not a cause of my PR, as it already happens in the `main` branch
image

File: chart/files/pod-template-file.kubernetes-helm-yaml [hunk 1]
```diff
@@ -35,15 +35,16 @@ metadata:
...
-    {{- if .Values.workers.podAnnotations }}
-      {{- toYaml .Values.workers.podAnnotations | nindent 4 }}
+    cluster-autoscaler.kubernetes.io/safe-to-evict: {{ .Values.workers.safeToEvict | quote}}
+    {{- if or .Values.airflowPodAnnotations .Values.workers.podAnnotations }}
+      {{- if .Values.airflowPodAnnotations }}
+        {{- toYaml .Values.airflowPodAnnotations | nindent 4 }}
+      {{- end }}
+      {{- if .Values.workers.podAnnotations }}
+        {{- toYaml .Values.workers.podAnnotations | nindent 4 }}
+      {{- end }}
```
Comment (romsharon98) [inline chart/files/pod-template-file.kubernetes-helm-yaml]: If `.Values.airflowPodAnnotations` and `.Values.workers.podAnnotations` have the same key with different values what will happen?

u can look in worker-deployment how it's been soved
https://github.com/apache/airflow/blob/65642ed9503f6135f781ac5e5158d375fb852d4a/chart/templates/workers/worker-deployment.yaml#L42

**Comment by eladkal** [comment]
@csp33 are you still working on this PR?

**Comment by csp33** [comment]
hey @eladkal , sorry for the delay. Comments have been addressed :)

**Review by romsharon98 — APPROVED** [review]

**Comment by csp33** [comment]
@jedcunningham ready for review :)

**Comment by csp33** [comment]
@potiuk @jedcunningham can you review this? 
thanks :)

**Comment by csp33** [comment]
> Looks good, but I think it needs a helm unit test.

@potiuk done!

**Review by potiuk — APPROVED** [review]
LGTM - @jedcunningham ?

**Comment by potiuk** [comment]
And @hussein-awala ?

**Review by eladkal — APPROVED** [review]

**Comment by ldacey** [comment]
> hi @tsgit , this behavior is not a caused by PR, as it already happens in the `main` branch 

Is there a workaround for the duplicate annotation? Unable to deploy 1.14.0 with ArgoCD due to duplicate keys.

**Comment by potiuk** [comment]
> > hi @tsgit , this behavior is not a caused by PR, as it already happens in the `main` branch
> 
> Is there a workaround for the duplicate annotation? Unable to deploy 1.14.0 with ArgoCD due to duplicate keys.

Copy and modify the helm chart locally and install it from there (and later contribute it back as a fix) ?

**Comment by eladkal** [comment]
> > hi @tsgit , this behavior is not a caused by PR, as it already happens in the `main` branch
> 
> Is there a workaround for the duplicate annotation? Unable to deploy 1.14.0 with ArgoCD due to duplicate keys.

See https://github.com/apache/airflow/pull/40554

**Pull request merged**